invalid memory address or nil pointer dereference when trying to import an archive file

## Describe the bug

I am trying to import my `.vim` directory as an archive.  I created a `vim.tar.gz` which looks like this:
```
$ tar --list -f ~/tmp/vim.tar.gz 
vim/colors/blackboard.vim
vim/
vim/ftdetect/vimwiki.vim
vim/spell/
vim/spell/en.utf-8.add.spl
vim/colors/calmar256-dark.vim
vim/skeleton/go.vim
vim/ftdetect/perl.vim
vim/ftdetect/pom.vim
vim/ftdetect/md.vim
vim/spell/en.latin1.add.spl
vim/ftdetect/cpp.vim
vim/colors/wombat256mod.vim
vim/ftdetect/yaml.vim
vim/colors/railscasts.vim
vim/ftdetect/
vim/colors/wombat256.vim
vim/skeleton/java.vim
vim/skeleton/
vim/colors/desert256.vim
vim/colors/peaksea.vim
vim/colors/lanzarotta.vim
vim/ftdetect/go.vim
vim/skeleton/perl.vim
vim/autoload/
vim/skeleton/c.vim
vim/colors/google.vim
vim/colors/
vim/ftdetect/java.vim
vim/spell/en.latin1.add
vim/skeleton/cpp.vim
vim/ftdetect/c.vim
vim/spell/en.utf-8.add
vim/colors/distinguished.vim
vim/autoload/plug.vim
```

Upon trying to import, I get the following error:
```
$ chezmoi import --verbose --strip-components 1 --destination ~/.vim --remove-destination --exact ~/tmp/vim.tar.gz
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xccbe66]

goroutine 1 [running]:
github.com/twpayne/chezmoi/cmd.(*Config).runImportCmd(0x18eaca0, 0x18e6d40, 0xc0002a52d0, 0x1, 0x7, 0x0, 0x0)
	/home/runner/work/chezmoi/chezmoi/cmd/import.go:79 +0x336
github.com/spf13/cobra.(*Command).execute(0x18e6d40, 0xc0002a5260, 0x7, 0x7, 0x18e6d40, 0xc0002a5260)
	/home/runner/go/pkg/mod/github.com/twpayne/cobra@v0.0.6/command.go:829 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0x18e7240, 0x0, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/twpayne/cobra@v0.0.6/command.go:917 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/twpayne/cobra@v0.0.6/command.go:867
github.com/twpayne/chezmoi/cmd.Execute()
	/home/runner/work/chezmoi/chezmoi/cmd/root.go:129 +0x2d
main.main()
	/home/runner/work/chezmoi/chezmoi/main.go:10 +0x20
```

If I change the destination from `~/.vim` to just `.vim`, I get this output instead.
```
$ chezmoi import --verbose --strip-components 1 --remove-destination --exact --destination .vim ~/tmp/vim.tar.gz
chezmoi: Rel: can't make .vim relative to /home/quincy
```


## Output of `chezmoi doctor`

```
$ chezmoi doctor
     ok: version 1.7.12, commit bed22fcb91787c0d3d2d8ca61ebfe7799380f961, built at 2020-01-24T02:36:32Z
     ok: runtime.GOOS linux, runtime.GOARCH amd64
     ok: /home/quincy/.local/share/chezmoi (source directory, perm 700)
     ok: /home/quincy (destination directory, perm 755)
     ok: /home/quincy/.config/chezmoi/chezmoi.toml (configuration file)
     ok: /bin/bash (shell)
     ok: /usr/local/bin/vim (editor)
     ok: /usr/local/bin/vimdiff (merge command)
     ok: /usr/bin/git (source VCS command, version 2.20.1)
     ok: /usr/bin/gpg (GnuPG, version 2.2.12)
warning: op (1Password CLI, not found)
warning: bw (Bitwarden CLI, not found)
warning: gopass (gopass CLI, not found)
warning: keepassxc-cli (KeePassXC CLI, not found)
     ok: /usr/bin/lpass (LastPass CLI, version 1.3.1)
warning: pass (pass CLI, not found)
warning: vault (Vault CLI, not found)
```

Heard you on FLOSS Weekly.  Can't wait to get it set up and working for me.

--- a/internal/chezmoi/targetstate.go
+++ b/internal/chezmoi/targetstate.go
@@ -151,10 +151,7 @@ func (ts *TargetState) Add(fs vfs.FS, addOptions AddOptions, targetPath string, 
 			if err != nil {
 				return err
 			}
-			if entry != nil {
-				return mutator.RemoveAll(filepath.Join(ts.SourceDir, entry.SourceName()))
-			}
-			return nil
+			return mutator.RemoveAll(filepath.Join(ts.SourceDir, entry.SourceName()))
 		}
 		contents, err := fs.ReadFile(targetPath)
 		if err != nil {
@@ -693,7 +690,11 @@ func (ts *TargetState) findEntry(name string) (Entry, error) {
 	if err != nil {
 		return nil, err
 	}
-	return entries[names[len(names)-1]], nil
+	entry, ok := entries[names[len(names)-1]]
+	if !ok {
+		return nil, os.ErrNotExist
+	}
+	return entry, nil
 }
 
 func (ts *TargetState) importHeader(r io.Reader, importTAROptions ImportTAROptions, header *tar.Header, mutator Mutator) error {

